index.js

```diff
@@ -631,6 +638,40 @@ Tome.prototype.getParent = function () {
 	return this.__parent__;
 };
 
+Tome.prototype.takeSnapshot = function () {
+	if (this.__root__ !== this) {
+		throw new Error('Only root tomes can take snapshots');
+	}
+
+	return {
+		content: Tome.unTome(this),
+		diff: this.__diff__.slice(),
+		version: this.__version__
+	};
+};
+
+Tome.prototype.restoreSnapshot = function (snapshot) {
+	if (this.__root__ !== this) {
+		throw new Error('Only root tomes can restore snapshots');
+	}
+
+	if (this.__version__ === snapshot.version) {
+		// nothing would change from restoring
+		return;
+	}
+
+	this.__version__ = snapshot.version;
+	this.__diff__ = snapshot.diff.slice();
+
+	this.__diffEnabled__ = false;
+
+	try {
+		this.assign(snapshot.content);
```

Comment:
Ah, but the problem with that is that we do emit events during `markDirty`, and if anyone would ever use this event to start restoring other snapshots this global would be screwed. So nevermind.